utt/privacy-wallet-service/src/PrivacyService.cpp

```diff
@@ -275,14 +275,16 @@ ::grpc::Status PrivacyWalletServiceImpl::handleUserClaimCoinsRequest(::grpc::Ser
       response->set_err(err_msg);
       return grpc::Status(grpc::StatusCode::ABORTED, err_msg);
     }
+  } catch (const libutt::api::operations::InvalidCoinsInTransfer& e) {
+    std::cout << e.what() << std::endl;
+    response->mutable_claim_coins_response()->set_warning(e.what());
```

Comment:
In this case, shouldn't we return false in the succ?
Also, I would suggest returning the invalid nullifiers and the reply.
Finally, we need to expose a way for the user to remove these invalid coins, otherwise it won't be able to proceed (the invalid coins will be chose over and over again)

Reply:
> In this case, shouldn't we return false in the succ?
Technically, the recipent was able to claim coins received. Reporting back false after claiming all valid coins seems incorrect to me. The more correct approach seems to be to return true with a warning.

>  Also, I would suggest returning the invalid nullifiers and the reply.
I didn't want the dapp to handle the invalid coins, but this is something that can be implemented. I'll discuss with you in more detail.